client/app/components/authors/SettingsPage.jsx

```diff
@@ -25,11 +25,21 @@ import "./SettingsPage.scss";
 import ErrorToast from "../shared/ErrorToast";
 
 const SettingsPage = ({
-    author, authorCredentialsUrl, customDomainIP, guestbookEntries, posts,
+    author, authorCredentialsUrl, customDomainIP, guestbookEntries, posts, scrollToGuestbook,
 }) => {
     const [isErrorToastDisplayed, setIsErrorToastDisplayed] = useState(false);
     const [errorToastMessage, setErrorToastMessage] = useState("");
 
+    if (scrollToGuestbook) {
+        useEffect(() => {
```

Comment:
@mobitar The `if` condition should go within the `useEffect`. If `scrollToGuestbook` changes, the `useEffect`'s function will be executed.
Also, hooks shouldn't be within an `if condition` https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level
```suggestion
    useEffect(() => {
        if (scrollToGuestbook) {
            setTimeout(() => {
                document.getElementById("guestbook-entries").scrollIntoView({
                    behavior: "smooth",
                });
            }, 250);
        }
    }, [scrollToGuestbook]);
```

Reply:
Good call! First time using hooks :)

Reply:
Also, what do you think about the setTimeout? I tried to avoid it, but for some reason without it, it would scroll to the guestbook section quickly, but then it would scroll back up to the top of the page. So it wasn't executing after the right time apparently.

Reply:
Yeah, I'm not a huge fan of the setTimeout, give me a sec, I'll debug this on my local machine and see how we can fix it :) @mobitar

Reply:
@mobitar I tested the behavior with and without the setTimeout and they both behave kind of weird. If you refresh the page it goes down and then back to the top. But if you open the link (no refresh) with and without the setTimeout it looks good. Should we remove the set timeout? When you test on your machine does it behave like that too? Maybe we can deploy and see how it looks in dev
Also, if you want to, another option (if it looks bad on dev without the setTimeout) we can create a ticket to replace all the "scroll into view" with a fancier implementation with hooks and find a way to fix this